### PR TITLE
Chapter1/homework/3

### PR DESCRIPTION
**Pull request by ShadowEnder**
Calculate BMI function written to calculate with meters instead of cm in order to pass tests.
Is calculated with cm in Demo function in the BMI variable (height / 100).

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            return Console.ReadLine();
+        }
+
+        public static int getInt(string message)
+        {
+            Console.Write(message);
+            return Convert.ToInt32(Console.ReadLine());
+        }
+
+        public static float getFloat(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'getFloat' to match pascal case naming rules, consider using 'GetFloat'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604&bid=16131167)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [The Cyclomatic Complexity of this method is 11 which is greater than 10 authorized.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                    calculateAgain = false;
+                }
+
+            }
+
+            Console.WriteLine("Exiting...");
+                        
+        }
+                
+        public static string getString(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'getString' to match pascal case naming rules, consider using 'GetString'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            Console.Write(message);
+            return Console.ReadLine();
+        }
+
+        public static int getInt(string message)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'getInt' to match pascal case naming rules, consider using 'GetInt'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            return Convert.ToInt32(Console.ReadLine());
+        }
+
+        public static float getFloat(string message)
+        {
+            Console.Write(message);
+            return Convert.ToSingle(Console.ReadLine());
+        }
+
+        public static float calculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'calculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 7]
```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604&bid=16131167)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [The Cyclomatic Complexity of this method is 11 which is greater than 10 authorized.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            return Convert.ToInt32(Console.ReadLine());
+        }
+
+        public static float GetFloat(string message)
+        {
+            Console.Write(message);
+            return Convert.ToSingle(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604&bid=16131167)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            return weight / (Convert.ToSingle(Math.Pow(height, 2)));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: This is a quite expension thing you do here. weight / (height * height) would be way better!

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+        public static int GetInt(string message)
+        {
+            Console.Write(message);
+            return Convert.ToInt32(Console.ReadLine());
+        }
+
+        public static float GetFloat(string message)
+        {
+            Console.Write(message);
+            return Convert.ToSingle(Console.ReadLine());
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Consider using float.Parse() instead.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+        public static string GetString(string message)
+        {
+            Console.Write(message);
+            return Console.ReadLine();
+        }
+
+        public static int GetInt(string message)
+        {
+            Console.Write(message);
+            return Convert.ToInt32(Console.ReadLine());
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Consider using int.Parse() instead.

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+            bool calculateAgain = true;
+
+            while (calculateAgain)
+            {
+                Console.WriteLine("BMI Calculator");
+
+                string name = GetString("Enter your name: ");
+                int age = GetInt("Enter your age: ");
+                float weight = GetFloat("Enter your weight(kg): ");
+                float height = GetFloat("Enter your height(cm): ");
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Get is a rather bad name for what those functions do. Prompt (as in Checks.cs) are a proper name for requested input.

File: Src/BootCamp.Chapter/Checks.cs [hunk 15]
```diff
@@ -17,25 +17,25 @@ public static class Checks
         public static int PromptInt(string message)
         {
             // To do: call your implementation. 
-            return 0;
+            return Lesson3.GetInt(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: Remove the TODO comments if you implemented them.

**Comment by ShadowEnder** [comment]
Codacy is complaining to changed the name of the function 'CalculateBMI' to 'CalculateBmi' to match PascalCase naming rules, but I'm not going to do that since it doesn't make sense to in this context.

Feedback about the homework, I didn't like that the tests run a value in meters when we were told for 1 & 2 to calculate it using cm (as in user enters their height in cm, which would then be passed to the CalculateBMI function but here the test is enter a value in meters which means you have to change the formula used in 1 & 2 to accommodate the test, which seems pointless since that's not what's supposed to be passed to the function.).

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+                Console.WriteLine(name + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " cm.");
+
+                float BMI = CalculateBMI(weight, height / 100);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Convention is to use bmi and CalculateBmi here.
You may ask yourself why CalculateBmi:
CalculateBMIFromPerson would be troublesome then.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863604&bid=16131167)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+                Console.WriteLine(name + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " cm.");
+
+                float BMI = CalculateBmi(weight, height / 100);
+
+                Console.WriteLine("Your BMI is " + BMI);
+
+                WeightType(BMI);                
+
+                Console.WriteLine("Do you want to calculate another persoon's BMI? (y/n)");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Response validation could also go into the function of its own.

**Pull request closed without merging**